### PR TITLE
Gb/regularization

### PR DESCRIPTION
**Pull request by grantbuster**

### CODE REVIEW

File: phygnn/phygnn.py [hunk 1]
```diff
@@ -142,8 +182,37 @@ def _preflight_p_fun(self, x, y_true, p, p_kwargs):
 
         logger.debug('p_fun passed preflight check.')
 
-    def _preflight_data(self, x, y, p):
-        """Run simple preflight checks on data shapes."""
+    def preflight_data(self, x, y, p):
```
Comment (MRossol) [inline phygnn/phygnn.py:185]: Why'd you decided to "expose" these methods?

File: phygnn/pre_processing.py [hunk 2]
```diff
@@ -68,46 +115,128 @@ def process_one_hot(self, convert_int=False):
...
                     oh_obj = OneHotEncoder(sparse=False)
+
                 oh_obj.fit(col)
                 one_hot_data.append(oh_obj.transform(col))
+
+        return one_hot_ind, one_hot_data, numerical_ind
+
+    def _make_df_one_hot_cols_labels(self, one_hot_ind, one_hot_data,
+                                     categories=None):
+        """
```
Comment (MRossol) [inline phygnn/pre_processing.py:139]: Add doc string

File: phygnn/phygnn.py (hunk 1 above)
Comment (grantbuster) [reply phygnn/phygnn.py:185]: People using this are _probably_ fairly competent and should be able to preflight their data before running training or predictions

File: phygnn/pre_processing.py (hunk 2 above)
Comment (grantbuster) [reply phygnn/pre_processing.py:139]: oops!

**Pull request merged**